src/main/java/wolox/training/service/OpenLibraryService.java

```diff
@@ -0,0 +1,67 @@
+package wolox.training.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import wolox.training.models.Book;
+
+@Service
+public class OpenLibraryService {
+
+  @Value("${openlibrary.url}")
+  private String base_uri;
+
+  private RestTemplate restTemplate = new RestTemplate();
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  public Optional<Book> bookInfo(String isbn) throws JsonProcessingException {
+    JsonNode jsonBook = getBookFromExternalAPI(isbn);
+    if(jsonBook.isEmpty()){
+      return Optional.empty();
+    }
```

Comment:
`return jsonBook.isEmpty() ? Optional.empty() : Optional.of(parseBookResponse(jsonBook));`

Reply:
`parseBookResponse` would have all the logic below the if